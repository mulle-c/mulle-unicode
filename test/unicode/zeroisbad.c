#include <mulle-utf/mulle-utf.h>

#include <stdio.h>
#include <stdlib.h>



static void  test( char *name,
                   int (*member_f)( mulle_utf32_t))
{
   printf( "%s zero %s member\n", name, (*member_f)( 0) ? "IS" : "is not");
}


int  main()
{
   test( "alphanumeric", mulle_utf32_is_alphanumeric);
   test( "capitalized", mulle_utf32_is_capitalized);
   test( "control", mulle_utf32_is_control);
   test( "decimaldigit", mulle_utf32_is_decimaldigit);
   test( "decomposable", mulle_utf32_is_decomposable);
   test( "legalcharacter", mulle_utf32_is_legalcharacter);
   test( "letter", mulle_utf32_is_letter);
   test( "lowercase", mulle_utf32_is_lowercase);
   test( "newline", mulle_utf32_is_newline);
   test( "nonbase", mulle_utf32_is_nonbase);
   test( "noncharacter", mulle_utf32_is_noncharacter);
   test( "nonpercentescape", mulle_utf32_is_nonpercentescape);
   test( "privatecharacter", mulle_utf32_is_privatecharacter);
   test( "punctuation", mulle_utf32_is_punctuation);
   test( "symbol", mulle_utf32_is_symbol);
   test( "uppercase", mulle_utf32_is_uppercase);
   test( "validurlfragment", mulle_utf32_is_validurlfragment);
   test( "validurlhost", mulle_utf32_is_validurlhost);
   test( "validurlpassword", mulle_utf32_is_validurlpassword);
   test( "validurlpath", mulle_utf32_is_validurlpath);
   test( "validurlquery", mulle_utf32_is_validurlquery);
   test( "validurlscheme", mulle_utf32_is_validurlscheme);
   test( "validurluser", mulle_utf32_is_validurluser);
   test( "whitespace", mulle_utf32_is_whitespace);
   test( "whitespaceornewline", mulle_utf32_is_whitespaceornewline);
   return( 0);
}

