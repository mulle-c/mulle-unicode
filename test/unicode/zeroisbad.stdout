alphanumeric zero is not member
capitalized zero is not member
control zero is not member
decimaldigit zero is not member
decomposable zero is not member
legalcharacter zero is not member
letter zero is not member
lowercase zero is not member
newline zero is not member
nonbase zero is not member
noncharacter zero is not member
nonpercentescape zero is not member
privatecharacter zero is not member
punctuation zero is not member
symbol zero is not member
uppercase zero is not member
validurlfragment zero is not member
validurlhost zero is not member
validurlpassword zero is not member
validurlpath zero is not member
validurlquery zero is not member
validurlscheme zero is not member
validurluser zero is not member
whitespace zero is not member
whitespaceornewline zero is not member
